File: Backend/src/main/java/com/vhu/backend/service/ArticleService.java
package com.vhu.backend.service;

import com.vhu.backend.dto.article.request.ArticleCreateRequest;
import com.vhu.backend.dto.article.request.ArticleUpdateRequest;
import com.vhu.backend.dto.article.response.ArticleResponse;
import com.vhu.backend.dto.article.response.ArticleTitleResponse;
import com.vhu.backend.entity.Article;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.Locale;

public interface ArticleService {
    Page<ArticleResponse> getAllArticles(int page, int size, String search);
    ArticleResponse togglePinStatus(Long articleId);
    ArticleResponse getArticleBySlug(String slug);
    ArticleResponse createArticle(ArticleCreateRequest request);
    ArticleResponse updateArticle(Long articleId, ArticleUpdateRequest request);
    void deleteArticle(Long articleId);
    List<ArticleTitleResponse> getAllArticleTitles();
    ArticleResponse getArticleById(Long articleId, Locale locale);

    List<ArticleResponse> getPinnedArticles(String languageCode, int limit);

    List<ArticleResponse> getLatestArticles(String languageCode, int limit);

    List<ArticleResponse> searchPublicArticles(String query, String languageCode);

    List<ArticleResponse> getLatestArticlesByCategoryId(Integer categoryId, String languageCode, int limit);

    List<ArticleResponse> getLatestArticlesByCategorySlug(String slug, String languageCode, int limit);

    List<ArticleResponse> getRelatedArticles(Long articleId, Integer categoryId, String languageCode, int limit);

    ArticleResponse mapToArticleResponse(Article article, String languageCode);
}

File: Frontend/admin/src/services/articleService.js
import apiClient from './api';
export const getAllArticles = (page = 0, size = 10, search = '') => {
    return apiClient.get(`/articles?page=${page}&size=${size}&search=${search}`);
};

export const getArticleById = (id) => apiClient.get(`/articles/id/${id}`);

export const createArticle = (articleData) => apiClient.post('/articles', articleData);

export const updateArticle = (id, articleData) => apiClient.put(`/articles/${id}`, articleData);

export const deleteArticle = (id) => apiClient.delete(`/articles/${id}`);

export const getAllArticleTitles = () => apiClient.get('/articles/list-titles');

export const toggleArticlePinStatus = (id) => {
    return apiClient.patch(`/articles/${id}/toggle-pin`);
};